mlfinlab/tests/test_sb_bagging.py

```diff
@@ -0,0 +1,310 @@
+                              sample_weight=np.ones((self.X_train.shape[0],)), )
+        with self.assertRaises(ValueError):
+            bagging_clf_4.fit(self.X_train, self.y_train_clf,
+                              sample_weight=np.ones((self.X_train.shape[0],)), )
+        with self.assertRaises(ValueError):
+            bagging_clf_5.fit(self.X_train, self.y_train_clf,
+                              sample_weight=np.ones((self.X_train.shape[0],)), )
+        with self.assertRaises(ValueError):
+            bagging_clf_6.fit(self.X_train, self.y_train_clf)
+            bagging_clf_6.n_estimators -= 2
+            bagging_clf_6.fit(self.X_train, self.y_train_clf)
+
+        with self.assertRaises(ValueError):
+            bagging_clf_7.fit(self.X_train, self.y_train_clf)
+            bagging_clf_7.n_estimators -= 1000
+            bagging_clf_7.fit(self.X_train, self.y_train_clf)
+
+    def test_sb_classifier(self):
+        """
+        Test Sequentially Bootstrapped Bagging Classifier
+        """
+
+        # Init classifiers
+        clf = RandomForestClassifier(n_estimators=1, criterion='entropy', bootstrap=False,
+                                     class_weight='balanced_subsample')
+
+        sb_clf = SequentiallyBootstrappedBaggingClassifier(base_estimator=clf, max_features=1.0, n_estimators=100,
+                                                           triple_barrier_events=self.meta_labeled_events,
+                                                           price_bars=self.data, oob_score=True, random_state=1)
+        sklearn_clf = BaggingClassifier(base_estimator=clf, max_features=1.0, n_estimators=50, oob_score=True,
+                                        random_state=1)
+
+        # X_train index should be in index mapping
+        self.assertTrue(self.X_train.index.isin(sb_clf.timestamp_int_index_mapping.index).all())
+
+        sb_clf.fit(self.X_train, self.y_train_clf)
+        sklearn_clf.fit(self.X_train, self.y_train_clf)
+
+        self.assertTrue((sb_clf.X_time_index == self.X_train.index).all())  # X_train index == clf X_train index
+
+        oos_sb_predictions = sb_clf.predict(self.X_test)
+        oos_sklearn_predictions = sklearn_clf.predict(self.X_test)
+
+        sb_precision = precision_score(self.y_test_clf, oos_sb_predictions)
+        sb_recall = recall_score(self.y_test_clf, oos_sb_predictions)
+        sb_roc_auc = roc_auc_score(self.y_test_clf, oos_sb_predictions)
+
+        sklearn_precision = precision_score(self.y_test_clf, oos_sklearn_predictions)
+        sklearn_recall = recall_score(self.y_test_clf, oos_sklearn_predictions)
+        sklearn_roc_auc = roc_auc_score(self.y_test_clf, oos_sklearn_predictions)
+
+        # Test OOB scores (sequentially bootstrapped, algorithm specific, standard (random sampling)
+
+        # Algorithm specific
+        self.assertGreater(sb_clf.oob_score_, sklearn_clf.oob_score_)  # oob_score for SB should be greater
```

Comment:
Hmm Why would it be higher? Wouldn't the sklearn oob score be inflated on account of training on redundant samples, thus inflating the oob scores. Ref chapter 6.

Reply:
Yeah, it is just for this example, because test classifier doesn't capture any pattern and train data set is very small